classes/TaskRunner/Miscellaneous/UpdateConfig.php

```diff
@@ -174,4 +182,57 @@ private function writeConfig(array $config): bool
 
         return (new UpgradeConfigurationStorage($this->container->getProperty(UpgradeContainer::WORKSPACE_PATH) . DIRECTORY_SEPARATOR))->save($this->container->getUpgradeConfiguration(), UpgradeFileNames::CONFIG_FILENAME);
     }
+
+    /**
+     * @throws ZipActionException
+     * @throws Exception
+     */
+    private function extractPrestashopVersionFromZip(string $zipFile): string
+    {
+        $internalZipFileName = 'prestashop.zip';
+        $versionFile = 'install/install_version.php';
+
+        if (!file_exists($zipFile)) {
+            throw new FileNotFoundException("Unable to find $zipFile file");
+        }
+        $zip = $this->container->getZipAction()->open($zipFile);
+        $internalZipContent = $this->container->getZipAction()->extractFileFromArchive($zip, $internalZipFileName);
+        $zip->close();
+
+        $tempInternalZipPath = $this->createTemporaryFile($internalZipContent);
+
+        $internalZip = $this->container->getZipAction()->open($tempInternalZipPath);
+        $fileContent = $this->container->getZipAction()->extractFileFromArchive($internalZip, $versionFile);
+        $internalZip->close();
+
+        unlink($tempInternalZipPath);
+
+        return $this->extractVersionFromContent($fileContent);
+    }
+
+    /**
+     * @throws IOException
+     */
+    private function createTemporaryFile(string $content): string
+    {
+        $tempFilePath = tempnam(sys_get_temp_dir(), 'internal_zip_');
+        if (file_put_contents($tempFilePath, $content) === false) {
+            throw new IOException('Unable to create temporary file');
+        }
+
+        return $tempFilePath;
+    }
+
+    /**
+     * @throws ZipActionException
+     */
+    private function extractVersionFromContent(string $content): string
+    {
+        $pattern = "/define\\('_PS_INSTALL_VERSION_', '([\\d.]+)'\\);/";
+        if (preg_match($pattern, $content, $matches)) {
+            return $matches[1];
+        } else {
+            throw new ZipActionException('Unable to extract version from content');
```

Comment:
Is the type `ZipActionException` appropriate here?

Reply:
Maybe not